src/material/tabs/tab-group.ts

```diff
@@ -243,11 +248,26 @@ export abstract class _MatTabGroupBase extends _MatTabGroupMixinBase implements
         }
       }
 
-      this._subscribeToTabLabels();
       this._changeDetectorRef.markForCheck();
     });
   }
 
+  /** Listens to changes in all of the tabs. */
+  private _subscribeToAllTabChanges() {
+    // Since we use a query with `descendants: true` to pick up the tabs, we may end up
+    // catching some that are inside of nested tab groups so we need to filter them out manually.
+    this._allTabs.changes
+      .pipe(startWith(this._allTabs))
+      .subscribe((tabs: QueryList<MatTab>) => {
+        this._tabs.reset(tabs.filter(tab => {
+          // @breaking-change 10.0.0 Remove null check for `_closestTabGroup`
+          // once it becomes a required parameter in MatTab.
+          return !tab._closestTabGroup || tab._closestTabGroup === this;
```

Comment:
I was able to figure it out, but I think that the comments here could use some more clarity around what this is doing. I.e. how it is filtering out tab that are inside nested tab groups.